### PR TITLE
Add subscription modes for java client doc.

### PR DESCRIPTION
**Pull request by codelipenghui**
### Motivation

Add document for java client to describe how to use each subscription modes and the difference between each subscription modes.

### Modifications

Add subscription modes section for java client doc.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by codelipenghui** [comment]
@Jennifer88huang @Anonymitaet Please help review this PR, i copied the document to google doc: https://docs.google.com/document/d/1a0OD7azjAZpCmiYw9fV8kagtXSwTkD004oC3Mz636eU/edit?usp=sharing

**Comment by Jennifer88huang** [comment]
Good idea, I'll give comments in the google doc.

**Comment by Jennifer88huang** [comment]
@codelipenghui I've finished reviewing the file in Google doc, you can check my comments there. If there is any issue, feel free to contact me.

**Comment by codelipenghui** [comment]
@Jennifer88huang Thanks for the review, i have synced updates to this PR.

**Comment by codelipenghui** [comment]
@merlimat @sijie @jiazhai Please help review the document if you have time.

**Comment by codelipenghui** [comment]
run java8 tests

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -306,6 +306,185 @@ consumerBuilder
...
+("key-2", "message-2-3")
+("key-3", "message-3-1")
+("key-3", "message-3-2")
+("key-4", "message-4-1")
+("key-4", "message-4-2")
+```
+
+> Note:
+>
+> If a topic is a partitioned topic, the partitioned topic only has one active consumer, and messages of the partition are distributed to multiple consumers.
```
Comment (sijie) [inline site2/docs/client-libraries-java.md]: hmm I think the description here is misleading.

I think it should contain two points:

- each partition only has one active consumer
- messages of one partition only distributed to one consumer
- multiple partitions are distributed to multiple consumers

@codelipenghui right?

Comment (codelipenghui) [reply site2/docs/client-libraries-java.md]: yes.

File: site2/docs/client-libraries-java.md [hunk 2]
```diff
@@ -306,6 +306,185 @@ consumerBuilder
...
+  
+Consumer consumer2 = client.newConsumer()
+        .topic("my-topic")
+        .subscriptionName("my-subscription")
+        .subscriptionType(SubscriptionType.Key_Shared)
+        .subscribe()
+//Both consumer1 and consumer2 are active consumers.
+```
+
+`Key_Shared` subscription is like `Shared` subscription, all consumers can attach to the same subscription. But it is different from `Key_Shared` subscription, messages with the same key are delivered to only one consumer in order. The possible distribution of messages between different consumers(by default we do not know in advance which keys will be assigned to a consumer, but a key will only be assigned to a consumer at the same time. ) consumer1 will receive:
```
Comment (wolfstudy) [inline site2/docs/client-libraries-java.md]:
```suggestion
`Key_Shared` subscription is like `Shared` subscription, all consumers can attach to the same subscription. But it is different from `Key_Shared` subscription, messages with the same key are delivered to only one consumer in order. The possible distribution of messages between different consumers(by default we do not know in advance which keys will be assigned to a consumer, but a key will only be assigned to a consumer at the same time. ) .

consumer1 will receive:
```

Comment (wolfstudy) [reply site2/docs/client-libraries-java.md]: All the content is good, there is a suggestion on the format.

**Comment by codelipenghui** [comment]
run java8 tests

**Pull request merged**